integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/massindexing/MassIndexingConditionalExpressionsIT.java

```diff
@@ -276,17 +317,12 @@ public void rootIndexed_someSubclassesIndexed_requestMassIndexingOnIndexedSubcla
 		OrmUtils.withinSession( sessionFactory, session -> {
 			SearchSession searchSession = Search.session( session );
 			MassIndexer indexer = searchSession.massIndexer( H2_B_Indexed.class );
+			indexer.type( H2_B_Indexed.class ).reindexOnly( "e.rootNumber = 2" );
 
 			backendMock.expectWorks( H2_B_Indexed.NAME, DocumentCommitStrategy.NONE, DocumentRefreshStrategy.NONE )
-					.add( "13", b -> b
-							.field( "rootText", "text13" )
-							.field( "bText", "text13" ) )
 					.add( "14", b -> b
 							.field( "rootText", "text14" )
 							.field( "bText", "text14" ) )
-					.add( "15", b -> b
-							.field( "rootText", "text15" )
-							.field( "bText", "text15" ) )
```

Comment:
I believe you did not test the case that (in my opinion) would prevent us from implementing [HSEARCH-4252](https://hibernate.atlassian.net/browse/HSEARCH-4252). I described this case [here](https://hibernate.zulipchat.com/#narrow/stream/132092-hibernate-search-dev/topic/HSEARCH-499/near/243491237):

```
@Entity abstract class A {
@Id Long id
}

@Entity class B extends A {
String myProperty;
}

@Entity class C extends A {
String myProperty;
}

MassIndexer massIndexer = session.massIndexer(A.class);
massIndexer.type(A.class).reindexOnly("myProperty = 'foo'");
massIndexer.startAndWait();
```

As you argued the example above was invalid (because `myProperty` doesn't exist on `A`), you can alternatively test this:


And I think two other cases are worth testing:

```
@Entity abstract class A {
@Id Long id
}

@Entity class B extends A {
String myProperty;
}

@Entity class C extends A {
String myProperty;
}

MassIndexer massIndexer = session.massIndexer(A.class);
massIndexer.type(B.class).reindexOnly("myProperty = 'foo'");
massIndexer.type(C.class).reindexOnly("myProperty = 'foo'");
massIndexer.startAndWait();
```

And since I know you have an easy workaround to handle the case above, here is another one that IMO cannot be solved easily:

```
@Entity abstract inteface I {
String getMyProperty();
}
@Entity abstract class A {
@Id Long id
}

@Entity class B extends A implements I {
String myProperty;
@Override
String getMyProperty() {
    return myProperty;
}
}

@Entity class C extends A implements I {
String myProperty;
@Override
String getMyProperty() {
    return myProperty;
}
}

MassIndexer massIndexer = session.massIndexer(I.class);
massIndexer.type(I.class).reindexOnly("myProperty = 'foo'");
massIndexer.startAndWait();
```